cudapoa/samples/sample_cudapoa.cpp

```diff
@@ -124,22 +125,148 @@ void process_batch(Batch* batch, bool msa, bool print)
     }
 }
 
-void generate_window_data(const std::string& input_file, const int number_of_windows, const int max_sequences_per_poa,
-                          std::vector<std::vector<std::string>>& windows, BatchSize& batch_size)
+size_t estimate_max_poas(const BatchSize& batch_size, const bool banded_alignment, const bool msa_flag)
 {
-    parse_window_data_file(windows, input_file, number_of_windows); // Generate windows.
-    assert(get_size(windows) > 0);
+    int32_t matrix_sequence_dimension = banded_alignment ? CUDAPOA_BANDED_MAX_MATRIX_SEQUENCE_DIMENSION : batch_size.max_matrix_sequence_dimension;
+    int32_t matrix_graph_dimension    = banded_alignment ? batch_size.max_matrix_graph_dimension_banded : batch_size.max_matrix_graph_dimension;
+    int32_t max_nodes_per_window      = banded_alignment ? batch_size.max_nodes_per_window_banded : batch_size.max_nodes_per_window;
```

Comment:
Hey Ramin,
Is there a reason why we're using signed instead of unsigned?

Reply:
Good point, although for size types it makes more sense to use unsigned int, but to be consistent with [`BatchSize`](https://github.com/clara-parabricks/ClaraGenomicsAnalysis/blob/dev-v0.5.0/cudapoa/include/claragenomics/cudapoa/batch.hpp#L50), they're defined as `int32_t`
Now the question is why `BatchSize` uses signed integer? The main reason is there were a mixed set of signed and unsigned integers used in cudaPOA which all changed into a template type. Since we could cover unsigned cases with signed type, but not the other way around, all such data in cudaPOA were changed to signed integers.
so to keep it consistent with the rest of the code, let's leave it as it is.

Reply:
I just saw this discussion and would like to comment.
Ideally, we should use signed integers for everything that should represent an integer number - even if it cannot be less than zero. Exceptions are if we really need the extra bit of unsigned to represent the large values (and cannot just use a larger signed integer), use the variable for bit-operations, or explicitly want modulo 2^type_bits arithmetic.
In short, the main reasons are that arithmetic with signed integers is less error prone and that signed integers can be checked for overflow or negative value errors (sometimes they also give the compiler a tiny bit more freedom to optimize).
(See, e.g. http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-mix and following guidelines.)
Today we _almost_ had one of these signed-unsigned arithmetic bugs: https://github.com/clara-parabricks/ClaraGenomicsAnalysis/pull/422#discussion_r423660027

Reply:
Thanks. That makes sense. I was just curious about this since the conventions for signed/unsigned seemed to be a bit mixed (size_t, intX_t, uintX_t).

Reply:
Yes, unfortunately, we are not really consistent here (yet?), which is also dangerous. Nevertheless, I would prefer signed integers wherever possible (and avoid size_t).